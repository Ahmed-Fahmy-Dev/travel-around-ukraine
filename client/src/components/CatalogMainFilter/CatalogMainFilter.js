import React from 'react';
import { styled, Button, Box, Typography } from '@mui/material';
import Grid from '@mui/material/Unstable_Grid2';
import { CatalogFilterPrice, CatalogFilterTourists, CatalogFilterRegion } from '../index';

const FilterWrapper = styled(Box)(({ theme }) => ({
  display: 'flex',
  flexDirection: 'column',
  margin: '0 auto',
  width: 252,
  padding: 0,

  [theme.breakpoints.up('tablet')]: {
    width: 540,
  },

  [theme.breakpoints.up('laptop')]: {
    width: 235,
  },
}));

const ShowButton = styled(Button)(() => ({
  padding: '12px 85px',
  alignSelf: 'center',
  marginTop: 40,
  maxWidth: 200,
}));

const CatalogMainFilter = () => (
  <FilterWrapper>
    <Typography variant="h3">Filters</Typography>

    <Grid container columnSpacing={5} sx={{ padding: 0 }}>
      <Grid spacing={4} item xs={12} tablet={6} laptop={12}>
        <CatalogFilterPrice />
        <CatalogFilterTourists />
      </Grid>

      <Grid item xs={12} tablet={6} laptop={12}>
        <CatalogFilterRegion />
      </Grid>
    </Grid>

    <ShowButton>Show</ShowButton>
  </FilterWrapper>
);

export default CatalogMainFilter;
