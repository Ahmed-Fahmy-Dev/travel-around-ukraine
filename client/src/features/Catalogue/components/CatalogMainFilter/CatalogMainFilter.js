import React, { useEffect } from 'react';
import axios from 'axios';

import { useSelector, useDispatch, shallowEqual } from 'react-redux';
import { useLocation } from 'react-router-dom';
import { styled, Button, Typography, Stack } from '@mui/material';
import Grid from '@mui/material/Unstable_Grid2';
import CloseIcon from '@mui/icons-material/Close';
import { CatalogFilterPrice, CatalogFilterDuration, CatalogFilterSeason } from '..';
import {
  setIsFilter,
  // setClearDuration,
  // setAllSeasons,
  fetchFilteredTours,
  setFilterParams,
} from '../../../../store/slices/filterSlice/filterSlice';
// import axiosConfig from '../../../../axiosConfig';


const FilterWrapper = styled(Stack)(({ theme }) => ({
  margin: '0 auto',
  width: 252,
  padding: 0,

  [theme.breakpoints.up('tablet')]: {
    width: 540,
  },

  [theme.breakpoints.up('laptop')]: {
    width: 235,
  },
}));

// const ShowButton = styled(Button)(() => ({
//   padding: '12px 85px',
//   alignSelf: 'center',
//   marginTop: 40,
//   maxWidth: 200,
// }));

const ResetButton = styled((props) => (
  <Button
    variant="text"
    startIcon={
      <CloseIcon
        sx={{
          width: '14px',
          height: '14px',
          color: 'gray',
        }}
      />
    }
    {...props}
  />
))(({ theme }) => ({
  padding: '5px 15px',
  marginTop: '12px',
  border: 'none',
  background: 'none',
  fontWeight: 500,
  fontSize: 12,
  textTransform: 'none',
  color: theme.palette.text.primary,
  alignSelf: 'center',

  '&:hover': {
    background: 'none',
    filter: 'brightness(1.5)',
    color: theme.palette.text.primary,
  },
}));

const CatalogMainFilter = () => {
  const dispatch = useDispatch();
  const products = useSelector((state) => state.catalogue.products, shallowEqual);
  const allPrices = [...new Set(products.map((tour) => tour.currentPrice))].sort((a, b) => a - b);


  // const duration = useSelector((store) => store.filter.duration);
  // const seasons = useSelector((store) => store.filter.seasons.map((el) => (el ? el.toLowerCase() : el)));
  const filterParams = useSelector((store) => store.filter.filterParams);
  // console.log(filterParams);
  // const isFilter = useSelector((state) => state.filter.isFilter);
  const location = useLocation();

  useEffect(() => {
    if (location.search && Object.keys(filterParams).length === 0) {
      dispatch(setIsFilter(true));
      const queryParams = new URLSearchParams(location.search).entries();
      dispatch(setFilterParams(Object.fromEntries(queryParams)));
    } else {
      dispatch(setIsFilter(false));
      dispatch(setFilterParams({}));
    }
  }, [location.search]);

  useEffect(() => {
    // console.log(products)
    if (Object.keys(filterParams).length !== 0) {
      const queryString = new URLSearchParams(filterParams).toString();
      const newUrl = new URL(`?${queryString}`, window.location);
      if (window.location.href !== newUrl.href) {
        window.location.assign(newUrl);
        
      }
    }
  }, [filterParams]);

  const filterTours = async () => {
    const params = new URLSearchParams();
    if (allPrices.length > 0 && location.search) {
      const priceFrom = Number(filterParams.price_from);
      const priceTo = Number(filterParams.price_to);

      if (priceFrom && priceTo) {
        const filterPrices = allPrices.filter((price) => price >= priceFrom && price <= priceTo);
        params.set('currentPrice', filterPrices);
      }
    }
    // console.log(params.toString());

    if (params.toString()) {
      
      dispatch(setIsFilter(true));
      await dispatch(fetchFilteredTours(params));
      
    }
  };

  useEffect(() => {
    filterTours();
  }, [products, filterParams]);

  // const filterTours = async () => {
  //   const params = new URLSearchParams();
  //   params.set('currentPrice', filterPrices());
  //   if (duration.length > 0) {
  //     params.set('duration', duration);
  //   }
  //   if (seasons.length > 0) {
  //     params.set('season', seasons.concat('all seasons'));
  //   }
  //   dispatch(setIsLoading(true));
  //   dispatch(setIsFilter(true));
  //   await dispatch(fetchFilteredTours(params));
  //   dispatch(setIsLoading(false));
  // };

  // const resetFilter = () => {
  //   // dispatch(setIsFilter(false));
  //   // dispatch(setFilteredTours([]));
  //   // dispatch(setPrices([Math.min.apply(null, allPrices), Math.max.apply(null, allPrices)]));
  //   // dispatch(setClearDuration([]));
  //   // dispatch(setClearDuration([]));
  //   // dispatch(setAllSeasons([]));
  // };

  return (
    <FilterWrapper>
      <Typography variant="h3">Filter</Typography>

      <Grid container columnSpacing={5} sx={{ p: 0 }}>
        <Grid spacing={4} item xs={12} tablet={6} laptop={12}>
          <CatalogFilterPrice />
          <CatalogFilterDuration />
        </Grid>

        <Grid item xs={12} tablet={6} laptop={12}>
          <CatalogFilterSeason />
        </Grid>
      </Grid>

      {/* <ShowButton onClick={filterTours}>Filter</ShowButton> */}
      <ResetButton>Reset filter</ResetButton>
    </FilterWrapper>
  );
};

export default CatalogMainFilter;
