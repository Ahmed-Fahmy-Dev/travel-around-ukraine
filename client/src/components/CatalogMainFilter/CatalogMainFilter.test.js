import { render } from '@testing-library/react';
import CatalogMainFilter from './CatalogMainFilter';

jest.mock('@mui/material/Button', () => ({ children }) => <button>{children}</button>);
jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <h3>{children}</h3>);
jest.mock('@mui/material/Unstable_Grid2', () => ({ children }) => <div>{children}</div>);


describe('CatalogMainFilter snapshot test', () => {
  test('should CatalogTourCard match snapshot', () => {
    const { asFragment } = render(<CatalogMainFilter />);
    expect(asFragment()).toMatchSnapshot();
  });
});
