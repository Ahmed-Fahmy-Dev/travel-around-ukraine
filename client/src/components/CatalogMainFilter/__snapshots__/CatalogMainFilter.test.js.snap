// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogMainFilter snapshot test should CatalogTourCard match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Filter
    </h3>
    <div>
      <div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-expanded MuiAccordion-gutters css-18xlkcq-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-expanded="true"
            class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-1t26u27-MuiButtonBase-root-MuiAccordionSummary-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
            >
              <h3>
                Price
              </h3>
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-1cwf0m9-MuiAccordionDetails-root"
                  >
                    <div
                      class="MuiBox-root css-gg4vpm"
                    >
                      <div
                        class="MuiInputBase-root MuiInputBase-colorPrimary css-wk9lxf-MuiInputBase-root"
                      >
                        <input
                          class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
                          placeholder="10"
                          type="text"
                          value="200"
                        />
                      </div>
                      <div
                        class="MuiInputBase-root MuiInputBase-colorPrimary css-wk9lxf-MuiInputBase-root"
                      >
                        <input
                          class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
                          placeholder="1000"
                          type="text"
                          value="600"
                        />
                      </div>
                    </div>
                    <div
                      class="MuiBox-root css-14q2pzh"
                    >
                      <span
                        class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-187mznn-MuiSlider-root"
                      >
                        <span
                          class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
                        />
                        <span
                          class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
                          style="left: 19.19191919191919%; width: 40.40404040404041%;"
                        />
                        <span
                          class="MuiSlider-thumb MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium css-eg0mwd-MuiSlider-thumb"
                          data-focusvisible="false"
                          data-index="0"
                          style="left: 19.19191919191919%;"
                        >
                          <input
                            aria-label="Price range"
                            aria-orientation="horizontal"
                            aria-valuemax="1000"
                            aria-valuemin="10"
                            aria-valuenow="200"
                            aria-valuetext="200 €"
                            data-index="0"
                            max="1000"
                            min="10"
                            step="1"
                            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                            type="range"
                            value="200"
                          />
                        </span>
                        <span
                          class="MuiSlider-thumb MuiSlider-thumbColorPrimary MuiSlider-thumbSizeMedium css-eg0mwd-MuiSlider-thumb"
                          data-focusvisible="false"
                          data-index="1"
                          style="left: 59.5959595959596%;"
                        >
                          <input
                            aria-label="Price range"
                            aria-orientation="horizontal"
                            aria-valuemax="1000"
                            aria-valuemin="10"
                            aria-valuenow="600"
                            aria-valuetext="600 €"
                            data-index="1"
                            max="1000"
                            min="10"
                            step="1"
                            style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
                            type="range"
                            value="600"
                          />
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-expanded MuiAccordion-gutters css-18xlkcq-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-expanded="true"
            class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-1t26u27-MuiButtonBase-root-MuiAccordionSummary-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
            >
              <h3>
                Tourists
              </h3>
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-1cwf0m9-MuiAccordionDetails-root"
                  >
                    <div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="1"
                        >
                          1 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="2"
                        >
                          2 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="3"
                        >
                          3 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="4"
                        >
                          4 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="5"
                        >
                          5 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="6"
                        >
                          6 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="7"
                        >
                          7 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="8"
                        >
                          8 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="9"
                        >
                          9 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="10"
                        >
                          10 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                      <div>
                        <button
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard css-cter6j-MuiButtonBase-root-MuiToggleButton-root"
                          tabindex="0"
                          type="button"
                          value="0"
                        >
                          More than 10 pers.
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAccordion-root Mui-expanded MuiAccordion-gutters css-18xlkcq-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-expanded="true"
            class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-1t26u27-MuiButtonBase-root-MuiAccordionSummary-root"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
            >
              <h3>
                Region
              </h3>
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  class="MuiAccordion-region"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root css-1cwf0m9-MuiAccordionDetails-root"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <h3>
                        All regions
                      </h3>
                    </label>
                    <div
                      class="MuiBox-root css-agnjr6"
                    >
                      <label
                        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      >
                        <span
                          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                        >
                          <input
                            class="PrivateSwitchBase-input css-1m9pwf3"
                            data-indeterminate="false"
                            id="region1"
                            type="checkbox"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="CheckBoxOutlineBlankIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                        <h3>
                          region1
                        </h3>
                      </label>
                      <label
                        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      >
                        <span
                          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                        >
                          <input
                            class="PrivateSwitchBase-input css-1m9pwf3"
                            data-indeterminate="false"
                            id="region2"
                            type="checkbox"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="CheckBoxOutlineBlankIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                        <h3>
                          region2
                        </h3>
                      </label>
                      <label
                        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      >
                        <span
                          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                        >
                          <input
                            class="PrivateSwitchBase-input css-1m9pwf3"
                            data-indeterminate="false"
                            id="region3"
                            type="checkbox"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="CheckBoxOutlineBlankIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                        <h3>
                          region3
                        </h3>
                      </label>
                      <label
                        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      >
                        <span
                          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                        >
                          <input
                            class="PrivateSwitchBase-input css-1m9pwf3"
                            data-indeterminate="false"
                            id="region4"
                            type="checkbox"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="CheckBoxOutlineBlankIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                        <h3>
                          region4
                        </h3>
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button>
      Filter
    </button>
    <button>
      Reset filter
    </button>
  </div>
</DocumentFragment>
`;
